docs/user-guide/troubleshooting.rst

```diff
@@ -61,3 +61,38 @@ install`` tool is able to verify the server's SSL certificate with a trusted CA.
 Details on how to install a new trusted CA are outside the scope of this
 document.
 
+Missing metadata.json file
+--------------------------
+
+Symptom
+^^^^^^^
+
+Uploading a puppet module results in MissingModuleFile Exception raised.
+
+Problem
+^^^^^^^
+
+One possible problem is that the tar.gz file being uploaded does not contain metadata.json file.
+Another possible problem is presence of more than one directory (Puppet module) inside the archive.
+
+Solution
+^^^^^^^^
+
+Upload a module that adheres to the `3.6+ metadata guidlines
+<https://docs.puppetlabs.com/puppet/latest/reference/modules_publishing.html#publishing-modules-on-the-puppet-forge>`_.
+Another solution is to upload an archive that contains only one Puppet module.
```

Comment:
s/Another solution is to upload an archive that contains /Also ensure that an uploaded archive contains /